cmd/podman/machine/ssh.go

```diff
@@ -0,0 +1,70 @@
+// +build amd64,linux amd64,darwin arm64,darwin
+
+package machine
+
+import (
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/podman/v3/cmd/podman/registry"
+	"github.com/containers/podman/v3/pkg/domain/entities"
+	"github.com/containers/podman/v3/pkg/machine"
+	"github.com/containers/podman/v3/pkg/machine/qemu"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	sshCmd = &cobra.Command{
+		Use:   "ssh [options] NAME [COMMAND [ARG ...]]",
+		Short: "SSH into a virtual machine",
+		Long:  "SSH into a virtual machine ",
+		RunE:  ssh,
+		Args:  cobra.MinimumNArgs(1),
+		Example: `podman machine ssh myvm
+  podman machine ssh -e  myvm echo hello`,
+
+		ValidArgsFunction: completion.AutocompleteNone,
```

Comment:
You have the AutocompleteNone function set but this command accepts arguments. To fix this you should add this function in `cmd/podman/common/completion.go` and use it here:
```golang
// AutocompleteMachineSSH - Autocomplete machine ssh command.
func AutocompleteMachineSSH(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
	if len(args) == 0 {
		return nil, cobra.ShellCompDirectiveNoFileComp
	}
	return nil, cobra.ShellCompDirectiveDefault
}
```
Generally speaking you should not use `NAME` in the use line unless it is for the create command. The `NAME` keyword expects no completion. I recommend using `MACHINE` instead. Technically this should complete the vm names but I see that there is no list command yet so I do not think there is a way to provide the names to a completion function for now. This should be changed when the list command is added.

Reply:
tyvm!